test/integration_test/applicationbackup_test.go

```diff
@@ -629,3 +724,168 @@ func commonApplicationBackupScheduleTests(
 	}
 	deletePolicyAndApplicationBackupSchedule(t, namespace, policyName, scheduleName)
 }
+
+func applicationBackupSyncControllerTest(t *testing.T) {
+	var err error
+	backupLocationName := appKey + "-backup-location-sync"
+
+	// Create myqsl app deployment
+	appCtx := createApp(t, appKey)
+
+	// Create backup location on first cluster
+	backupLocation, err := createBackupLocation(t, backupLocationName, appCtx.GetID(), storkv1.BackupLocationS3, "secret-config")
+	require.NoError(t, err, "Error creating backuplocation")
+	logrus.Infof("Created backup location:%s sync:%t", backupLocation.Name, backupLocation.Location.Sync)
+	backupLocation.Location.Sync = true
+
+	firstBackup, err := createApplicationBackupWithAnnotation(t, appKey+"-backup-sync", appCtx.GetID(), backupLocation)
+	require.NoError(t, err, "Error creating app backups")
+
+	// Create backup location on second cluster
+	err = dumpRemoteKubeConfig(remoteConfig)
+	if err != nil {
+		require.NoErrorf(t, err, "Unable to write clusterconfig: %v", err)
+	}
+
+	err = setRemoteConfig(remoteFilePath)
+	require.NoError(t, err, "Error setting remote config")
+
+	// Create namespace for the backuplocation on second cluster
+	ns, err := k8s.Instance().CreateNamespace(appCtx.GetID(),
+		map[string]string{
+			"creator": "stork-test",
+			"app":     appCtx.App.Key,
+		})
+	require.NoError(t, err, "Failed to create namespace %s: %v", appCtx.GetID(), err)
+
+	backupLocation2, err := createBackupLocation(t, backupLocationName, ns.Name, storkv1.BackupLocationS3, "secret-config")
+	require.NoError(t, err, "Error creating backuplocation on second cluster")
+	logrus.Infof("Created application backup on second cluster %s: sync:%t", backupLocation.Name, backupLocation.Location.Sync)
+
+	// Set sync to true on second cluster so that backup location gets synced
+	backupLocation2.Location.Sync = true
+	_, err = k8s.Instance().UpdateBackupLocation(backupLocation2)
+	require.NoError(t, err, "Failed to set backup-location sync to true")
+	logrus.Infof("Updated application backup on 2nd cluster %s: sync:%t", backupLocation2.Name, backupLocation2.Location.Sync)
+
+	// Check periodically to see if the backup from this test is synced on second cluster
+	var allAppBackups *storkv1.ApplicationBackupList
+	listBackupsTask := func() (interface{}, bool, error) {
+		allAppBackups, err = k8s.Instance().ListApplicationBackups(ns.Name)
+		if err != nil || len(allAppBackups.Items) == 0 {
+			logrus.Infof("Failed to list app backups on second cluster. Error: %v. Number of backups: %v", err, len(allAppBackups.Items))
```

Comment:
`allAppBackups` will be nil if err is not nil, so this will hit a nil pointer exception in that case

Reply:
Fixed, will also check if allAppBackups is nil.

Reply:
The problem still exists. If err is not nil you are trying to print the length of `allAppBackups.Items` when `allAppBackups` is nil

Reply:
Fixed.